modules/sc-mesh-secure-deployment/src/nats/comms_nats_controller.py

```diff
@@ -468,12 +480,33 @@ def __action_certificates(self, response: requests.Response) -> str:
             certificates_dict = json.loads(response.text)["payload"]["certificates"]
 
             for certificate_name, certificate in certificates_dict.items():
-                with open(
-                    f"{self.__cbm_certs_downloaded}/{certificate_name}", "wb"
-                ) as cert_file:
-                    cert_file.write(base64.b64decode(certificate))
+                if str(certificate_name).endswith(".tar.bz2"):
+                    with open(
+                        f"{self.__cbm_certs_downloaded}/{certificate_name}", "wb"
+                    ) as cert_file:
+                        cert_file.write(base64.b64decode(certificate))
+                        self.__comms_controller.logger.debug(
+                            "Certificate %s saved", certificate_name
+                        )
+                        with tarfile.open(
+                            f"{self.__cbm_certs_downloaded}/{certificate_name}", "r:bz2"
+                        ) as tar:
+                            tar.extractall(path=cert_path)
+                            self.__comms_controller.logger.debug(
+                                "Certificate %s extracted to %s",
+                                certificate_name,
+                                cert_path,
+                            )
+                elif str(certificate_name).endswith("-sig"):
+                    with open(
+                        f"{self.__cbm_certs_downloaded}/{certificate_name}", "wb"
```

Comment:
Same as in the `.tar.bz2` comment above :face_in_clouds:

Reply:
Same as above:
certificate_name = os.path.basename(certificate_name)